### PR TITLE
Add MultiMed-ST S2T recipe with OWSM v4 small fine-tuning

### PR DESCRIPTION
**Pull request by KuangDW**
## What did you change?

This PR adds an ESPnet2 S2T recipe for English-to-German speech translation on the MultiMed-ST dataset.
The recipe includes:
- Data preparation for MultiMed-ST into ESPnet-style directories.
- OWSM-format `text`, `text.prev`, and `text.ctc` preparation.
- OWSM v4 small fine-tuning support using `espnet/owsm_v4_small_370M`.
- GPU decoding configuration for English-to-German speech translation.
- Local Stage 13 ST scoring with BLEU, chrF, and TER via `sacrebleu`.

---

## Why did you make this change?

MultiMed-ST is a large-scale multilingual medical speech translation dataset. Adding this recipe makes it easier to reproduce OWSM fine-tuning and evaluation for English-to-German medical speech translation.

Reference:
- MultiMed-ST: Large-scale Many-to-many Multilingual Medical Speech Translation (https://arxiv.org/abs/2504.03546)

---

## Is your PR small enough?

Mostly yes. The PR is limited to a new local recipe under `egs2/multimed_st/s2t1` and one entry in `egs2/README.md`.

It slightly exceeds the suggested guideline because ESPnet recipe scaffold files are included:
- 24 files changed
- 1052 insertions

The dataset-specific implementation is limited to:
- `run.sh`
- `db.sh`
- `local/data.sh`
- `local/prepare_multimed_st.py`
- `local/prepare_owsm_v4_assets.py`
- `local/score.sh`
- `conf/decode_owsm_st_de.yaml`

---

## Additional Context

The README includes both zero-shot and fine-tuned OWSM v4 small results on the MultiMed-ST English-to-German test set. ST metrics computed by `local/score.sh` after stripping OWSM prompt tokens:

| Model | BLEU | chrF2 | TER |
| --- | ---: | ---: | ---: |
| OWSM v4 small zero-shot | 29.4 | 56.3 | 60.6 |
| OWSM v4 small fine-tuned | 31.4 | 58.0 | 58.0 |

ESPnet default Stage 13 scoring is also reported in the README for reference. 

The generated fine-tuning config disables Flash Attention and fused AdamW by default, and uses a conservative learning rate for OWSM fine-tuning. To enable Flash Attention or a fused optimizer for faster training, modify `local/prepare_owsm_v4_assets.py`  or the generated `conf/finetune_owsm_v4_small.yaml`before training.

### CODE REVIEW

File: egs2/multimed_st/s2t1/setup.sh
Comment (sw005320) [inline egs2/multimed_st/s2t1/setup.sh:1]: I believe you don't need this script.

File: egs2/multimed_st/s2t1/conf/fbank.conf
Comment (sw005320) [inline egs2/multimed_st/s2t1/conf/fbank.conf:1]: This should be reverted.
We include the actual config file so users can customize it.

File: egs2/multimed_st/s2t1/conf/pbs.conf
Comment (sw005320) [inline egs2/multimed_st/s2t1/conf/pbs.conf:1]: Ditto
This should be reverted.
We include the actual config file so users can customize it.

File: egs2/multimed_st/s2t1/conf/pitch.conf
Comment (sw005320) [inline egs2/multimed_st/s2t1/conf/pitch.conf:1]: Ditto
This should be reverted.
We include the actual config file so users can customize it.

File: egs2/multimed_st/s2t1/conf/queue.conf
Comment (sw005320) [inline egs2/multimed_st/s2t1/conf/queue.conf:1]: Ditto
This should be reverted.
We include the actual config file so users can customize it.

File: egs2/multimed_st/s2t1/conf/slurm.conf
Comment (sw005320) [inline egs2/multimed_st/s2t1/conf/slurm.conf:1]: Ditto
This should be reverted.
We include the actual config file so users can customize it.

File: egs2/multimed_st/s2t1/cmd.sh
Comment (sw005320) [inline egs2/multimed_st/s2t1/cmd.sh:1]: Ditto
This should be reverted.
We include the actual config file so users can customize it.

**Review by sw005320 — COMMENTED** [review]
Some files should be actual files, not a symlink.

@Masao-Someki, can you also review this PR?

File: egs2/multimed_st/s2t1/local/prepare_multimed_st.py [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+        src_lang,
+        split=split,
+        streaming=True,
+    )
+    ds = ds.cast_column("audio", Audio(decode=False))
+    return iter(ds)
+
+
+def prepare_split(
+    *,
```
Comment (Masao-Someki) [inline egs2/multimed_st/s2t1/local/prepare_multimed_st.py]: We don't want this. Just a formatting but please remove it

File: egs2/multimed_st/s2t1/local/prepare_multimed_st.py [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+}
+
+LANG_CODE = {
+    "English": "eng",
+    "German": "deu",
+    "French": "fra",
+    "Vietnamese": "vie",
+    "Chinese": "zho",
+    "traditional_chinese": "zho_trad",
+}
```
Comment (Masao-Someki) [inline egs2/multimed_st/s2t1/local/prepare_multimed_st.py]: At least for espnet2, I think it is common to use abbreviations? Like `eng` or `deu` from iso language code

File: egs2/multimed_st/s2t1/local/score.sh [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        file=sys.stderr,
+    )
+
+with open(sys.argv[3], "w", encoding="utf-8") as f_ref, open(
+    sys.argv[4], "w", encoding="utf-8"
+) as f_hyp:
+    for key in keys:
+        f_ref.write(ref[key] + "\n")
+        f_hyp.write(hyp[key] + "\n")
+PY
```
Comment (Masao-Someki) [inline egs2/multimed_st/s2t1/local/score.sh]: Well, I think you can just write an another python script and call it here. It is simply difficult to read (because we don't have code highlight)

**Review by Masao-Someki — COMMENTED** [review]
Thanks for the PR!
I added comments but mostly it is formatting issue

File: egs2/multimed_st/s2t1/local/score.sh (hunk 3 above)
Comment (KuangDW) [reply egs2/multimed_st/s2t1/local/score.sh]: Thanks for the review! I updated the code to address the comments: removed the keyword-only marker, keyed the language tables by language codes, and moved the inline Python from `score.sh` to `local/align_ref_hyp.py`. Please let me know if I misunderstood anything.

File: egs2/multimed_st/s2t1/local/prepare_owsm_v4_assets.py
Comment (sw005320) [inline egs2/multimed_st/s2t1/local/prepare_owsm_v4_assets.py:1]: I like this script.
This script may be used for the other recipes.
We may maintain this in our common utility place (e.g., https://github.com/espnet/espnet/tree/master/utils or template directory) in the future.

File: egs2/multimed_st/s2t1/README.md
Comment (sw005320) [inline egs2/multimed_st/s2t1/README.md:1]: You may consider using OWSM v4 large models

**Comment by sw005320** [comment]
Thanks, @KuangDW!
Your bootcamp is done!
